File: 0007-reverse-integer/0007-reverse-integer.ts
function reverse(x: number): number {
    const neg = x < 0;

    let num = Math.abs(x);
    let rev = 0
    while(num > 0) {
        let ld = num % 10;
        rev = ld + (rev * 10);
        num = Math.floor(num / 10);
    }

    if(rev > 2**31 - 1) return 0;
    return neg ? -rev: rev;
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int n = x;
        if(n < 0) {
            n = -1 * x;
        }

        long rev = 0;
        while(n > 0) {
            int ld = n % 10;
            rev = ld + (rev * 10);
            n = n / 10;
        }

        if(rev < Integer.MIN_VALUE || rev > Integer.MAX_VALUE) return 0;

        if(x < 0) return (int) -rev;
        return (int) rev;
    }
}